### PR TITLE
Add types for `rockset` node module

### PR DESCRIPTION
**Pull request by kwadhwa18**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/rockset/tslint.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": { "no-empty-interface": false  }
```
Comment (armanio123) [inline types/rockset/tslint.json]: Is there a reason why you need this off?

File: types/rockset/rockset-tests.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { Collection } from "rockset";
+
+function rockset_tests() {
+    // TODO
```
Comment (armanio123) [inline types/rockset/rockset-tests.ts]: Could you please add a couple of tests before merging the pull request?

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/rockset/tslint.json (hunk 1 above)
Comment (kwadhwa18) [reply types/rockset/tslint.json]: I had few interfaces which are empty and are used in other interfaces. e.g. `AddDocumentsRequestData`. Is there any other way to resolve this?

File: types/rockset/rockset-tests.ts (hunk 2 above)
Comment (kwadhwa18) [reply types/rockset/rockset-tests.ts]: sure, we had a user waiting for the definitions so created a pull request without tests. Will add some.

File: types/rockset/tslint.json (hunk 1 above)
Comment (armanio123) [reply types/rockset/tslint.json]: The linter documentation states: `An empty interface is equivalent to its supertype (or {}).`

So basically it serves no purpose other than saying the type is an `object`. Instead you could just use:
```ts
export interface AddDocumentsResponse {
    data: {}[];
}
```

Comment (kwadhwa18) [reply types/rockset/tslint.json]: done

File: types/rockset/rockset-tests.ts (hunk 2 above)
Comment (kwadhwa18) [reply types/rockset/rockset-tests.ts]: Added some tests

**Comment by kwadhwa18** [comment]
The master branch seems to be failing causing Travis to fail for my pull request

**Review by armanio123 — APPROVED** [review]

**Pull request merged**